### PR TITLE
Fuzzer for bootspec.c and associated fixes

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

**Comment by bluca** [comment]
mkosi/tumbleweed: `/root/mkosi.build: line 70: groupadd: command not found`

**Comment by keszybz** [comment]
> mkosi/tumbleweed: `/root/mkosi.build: line 70: groupadd: command not found`

That's unrelated, right?

**Comment by bluca** [comment]
> > mkosi/tumbleweed: `/root/mkosi.build: line 70: groupadd: command not found`
> 
> That's unrelated, right?

yeah definitely look unrelated, but also new (to me at least)

File: src/shared/bootspec.h [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
 #include "set.h"
 #include "string-util.h"
+#include "json.h"
```
Comment (bluca) [inline src/shared/bootspec.h]: sorting?

File: src/shared/bootspec.c [hunk 2]
```diff
@@ -168,6 +168,25 @@ static int boot_entry_load_type1(
...
 }
 
+int boot_config_load_type1(
+                BootConfig *config,
+                FILE *f,
+                const char *root,
+                const char *dir,
+                const char *id) {
+        int r;
+
```
Comment (bluca) [inline src/shared/bootspec.c:185]: few asserts missing

**Review by bluca — APPROVED** [review]

**Comment by keszybz** [comment]
Thanks, updated.

**Comment by DaanDeMeyer** [comment]
> mkosi/tumbleweed: /root/mkosi.build: line 70: groupadd: command not found

cc @Werkov for the opensuse tumbleweed mkosi failure.

**Review by yuwata — APPROVED** [review]

**Pull request merged**